mlflow/models/model.py

```diff
@@ -307,3 +307,66 @@ def log(
                     await_registration_for=await_registration_for,
                 )
         return mlflow_model.get_model_info()
+
+
+def get_model_info(model_uri: str) -> ModelInfo:
+    """
+    Create a :py:class:`ModelInfo <mlflow.models.model.ModelInfo>` instance that contains the
+    model metadata from the logged model uri.
+
+    :param model_uri: The location, in URI format, of the MLflow model. For example:
+
+                      - ``/Users/me/path/to/local/model``
+                      - ``relative/path/to/local/model``
+                      - ``s3://my_bucket/path/to/model``
+                      - ``runs:/<mlflow_run_id>/run-relative/path/to/model``
+                      - ``models:/<model_name>/<model_version>``
+                      - ``models:/<model_name>/<stage>``
+                      - ``mlflow-artifacts:/path/to/model``
+
+                      For more information about supported URI schemes, see
+                      `Referencing Artifacts <https://www.mlflow.org/docs/latest/concepts.html#
+                      artifact-locations>`_.
+
+    :return: A :py:class:`ModelInfo <mlflow.models.model.ModelInfo>` instance that contains the
+            metadata of the logged model.
+    """
+    from mlflow.pyfunc import _download_artifact_from_uri
+
+    local_path = _download_artifact_from_uri(artifact_uri=model_uri, output_path=None)
+    model_meta = Model.load(os.path.join(local_path, MLMODEL_FILE_NAME))
+    return ModelInfo(
+        artifact_path=model_meta.artifact_path,
+        flavors=model_meta.flavors,
+        model_uri=model_uri,
+        model_uuid=model_meta.model_uuid,
+        run_id=model_meta.run_id,
+        saved_input_example_info=model_meta.saved_input_example_info,
+        signature_dict=model_meta.signature.to_dict() if model_meta.signature else None,
+        utc_time_created=model_meta.utc_time_created,
+        mlflow_version=model_meta.mlflow_version,
+    )
+
+
+def get_model_signature(model_uri: str) -> dict:
```

Comment:
Instead of having two methods, it would be great to include the model signature in the `ModelInfo` class so that users can call `get_model_info().signature`. I realize that `ModelInfo` has a `signature_dict` attribute; is it possible to add a `signature` attribute of type `ModelSignature` (rather than `dict`) and mark `signature_dict` as deprecated via the `@deprecated` decorator?

Reply:
Actually @depracted decorator can only be applied on classes or methods, so I added another similar decorator for this reference (let me know if you have other preferences since I'm not super expert here).